docs/development/documentation.md

```diff
@@ -0,0 +1,28 @@
+# Documentation
+
+Sphinx est utilisé pour la génération de la documentation depuis des pages écrites en Markdown (via le [parser MyST](https://myst-parser.readthedocs.io/en/latest/)).
+
+## Génération du site web de documentation
+
+Pour la génération:
+
+```bash
+# install base dependencies
+python -m pip install -U -r requirements/base.txt
+# install aditionnal dependencies
+python -m pip install -U -r requirements/documentation.txt
+# build it
+sphinx-build -b html docs docs/_build
```

Comment:
When I run this command in a container, I have this error 
```
# sphinx-build -b html docs docs/_build
Running Sphinx v5.3.0

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/sphinx/config.py", line 350, in eval_config_file
    exec(code, namespace)
  File "/home/docker/docs/conf.py", line 15, in <module>
    from r2gg import __about__
  File "/home/docker/r2gg/__init__.py", line 1, in <module>
    from r2gg._configure import configure
  File "/home/docker/r2gg/_configure.py", line 5, in <module>
    import psycopg2
  File "/usr/local/lib/python3.11/site-packages/psycopg2/__init__.py", line 51, in <module>
    from psycopg2._psycopg import (                     # noqa
SystemError: initialization of _psycopg raised unreported exception
```
I run inside it inside a `docker run -it -v /home/Documents/itineraire/route-graph-generator/:/home/docker/ python:bullseye bash`. It worked for road2 documentation. :/

Reply:
You need to install base requirements and documentation requirements before running this command. But there is a problem when I try to install base requirements. I'm looking for the reason...

Reply:
I did it (and I had an error with lxml). But this previous error was there. Thanks for looking at it.

Reply:
Actually, we need to refactor entry points of r2gg to be able to build documentation without installing base requirements (see #20 )

Reply:
ok !

Reply:
@lgrd you should now be able to build documentation without installing base requirements

Reply:
thanks ! it works ;)